bundles/mapping/infobox/plugin/openlayerspopup/OpenlayersPopupPlugin.ol3.js

```diff
@@ -658,36 +664,71 @@ Oskari.clazz.define(
          * Pans map if gfi popup would be out of screen
          * @param {Array} lonlat where to show the popup
          */
-        _panMapToShowPopup: function (lonlatArray) {
+        _panMapToShowPopup: function (lonlatArray, positioning, margins) {
+            var margins = margins || this._viewportMargins;
+            // don't try to pan the map if gfi popup position isn't in the viewport
+            if (!this.getMapModule().isLonLatInViewport(lonlatArray, margins.isInViewport)){
+                return;
+            }
+
             var me = this,
+                posClasses = me._positionClasses, // supported ol.OverlayPositionings
                 pixels = me.getMapModule().getPixelFromCoordinate(lonlatArray),
-                size = me.getMapModule().getSize(),
-                width = size.width - 128, // add some safety margin here so the popup close button won't go under the zoombar...
-                height = size.height - 128;
-            // if infobox would be out of screen
-            // -> move map to make infobox visible on screen
-            var panx = 0,
-                pany = 0,
+                mapSize = me.getMapModule().getSize(),
+                panY = 0,
+                panX = 0,
+                positionX = 'left', // default ('no-position-info')
+                positionY = 'top', // default ('no-position-info')
                 popup = jQuery('.olPopup'),
-                infoboxWidth = popup.width(),
-                infoboxHeight = popup.height();
-
-            if (pixels.x + infoboxWidth > width) {
-                if (infoboxWidth > width) {
-                    panx = -pixels.x;
-                } else {
-                    panx = width - (pixels.x + infoboxWidth);
+                popupX = popup.width(),
+                popupY = popup.height();
+
+            //If supported ol.OverlayPositioning is used, use it instead of default values
+            Object.keys(posClasses).forEach(function (pos){
+                if (positioning === posClasses[pos]){
+                    positionX = positioning.split("-")[1];
+                    positionY = positioning.split("-")[0];
```

Comment:
Split once and take the results:

    var popupDirection = positioning.split("-");
    positionY = popupDirection[0];
    positionX = popupDirection[1];